OpenRA.Mods.Common/Activities/Air/HeliAttack.cs

```diff
@@ -16,21 +19,27 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class HeliAttack : Activity
 	{
-		Target target;
-		public HeliAttack(Target target) { this.target = target; }
+		readonly Target target;
+		readonly Helicopter helicopter;
+		readonly AttackHeli attackHeli;
+		readonly IEnumerable<AmmoPool> ammoPools;
+
+		public HeliAttack(Actor self, Target target)
+		{
+			this.target = target;
+			helicopter = self.Trait<Helicopter>();
+			attackHeli = self.Trait<AttackHeli>();
+			ammoPools = self.TraitsImplementing<AmmoPool>();
+		}
 
 		public override Activity Tick(Actor self)
 		{
 			if (IsCanceled || !target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			var reloads = self.TraitOrDefault<Reloads>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo() && reloads == null)
+			if (ammoPools != null && ammoPools.All(x => !x.Info.SelfReloads && !x.HasAmmo()))
```

Comment:
Btw I changed this comment intentionally. As much as I want it to be generic as the above one on `FlyAttack`, this one explicitly orders the helicopter to return to base with `HeliReturn`.
Anyway, this is what the comment should be, so hopefully we'll fix the activity soon, so it matches the one in `FlyAttack`.